### PR TITLE
Adds Vagrant files for running Windows2016 or Windows2012 VM with puppet agent

### PR DESCRIPTION
**Pull request by diverdane**
### What does this PR do?
This PR adds Vagrantfiles, Bash scripts, and PowerShell scripts
that can be used to create a Vagrant/VirtualBox-based development and test
environment for testing the Conjur Puppet module with Puppet Agents running
on Windows2016 or Windows2012.

The Vagrantfiles and scripts can be used to:

- Spin up a containerized Puppet Server and Conjur server via docker-compose.
- Create a Windows2016 or Windows2012 VM.
- Dynamically install the desired version of Puppet Agent on the VM.
- Run Puppet Agent on the VM to install a Puppet catalog.
- Confirm that Puppet has been provisioned according to the configured
  Puppet manifest on the Puppet master.

For details, see the README.md in
examples/puppetmaster/vagrant/windows2016_puppet.

### What ticket does this PR close?
Issue #95

#### Change log
- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [x] The changes in this PR do not require tests

#### Documentation
- [x] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [ ] This PR does not require updating any documentation

### CODE REVIEW

File: examples/puppetmaster/vagrant/windows2016/README.md [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+### Clone this Repository
+
+Make a local copy of this repository as follows:
+
+```sh-session
+    git clone https://github.com/cyberark/conjur-puppet
+    cd examples/puppetmaster/vagrant/windows2016_puppet
+```
+
+### Setting up Vagrant
```
Comment (sgnn7) [inline examples/puppetmaster/vagrant/windows2016/README.md:72]: This doesn't say how to install vagrant, just how to install the plugin

File: examples/puppetmaster/vagrant/windows2016/README.md [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+
+## Create a Puppet Master
+
+We'll need a Puppet master to which our Puppet agent can connect.
+To do this, run the Puppet master smoketest, but leave the Puppet master
+and Conjur containers running:
+
+```sh-session
+    cd ../..
+    export CLEANUP_ON_EXIT=true
```
Comment (sgnn7) [inline examples/puppetmaster/vagrant/windows2016/README.md:103]: I think this needs to be false

Comment (sgnn7) [reply examples/puppetmaster/vagrant/windows2016/README.md:103]: Also seems to be `CLEAN_UP_ON_EXIT` not `CLEANUP_ON_EXIT`

File: examples/puppetmaster/vagrant/windows2016/README.md (hunk 1 above)
Comment (diverdane) [reply examples/puppetmaster/vagrant/windows2016/README.md]: I added the link to the Vagrant installation instructions to this section.

File: examples/puppetmaster/vagrant/windows2016/README.md (hunk 2 above)
Comment (diverdane) [reply examples/puppetmaster/vagrant/windows2016/README.md]: Fixed.

**Comment by diverdane** [comment]
@izgeri, @sgnn7 - I'm done with updating this, Jenkins tests are passing, and it's ready for re-review. Thanks!

**Pull request merged**